controller/konnect/reconciler_consumerref.go

```diff
@@ -103,15 +103,19 @@ func handleKongConsumerRef[T constraints.SupportedKonnectEntityType, TEnt constr
 		return ctrl.Result{Requeue: true}, nil
 	}
 
-	old := ent.DeepCopyObject().(TEnt)
-	if err := controllerutil.SetOwnerReference(&consumer, ent, cl.Scheme(), controllerutil.WithBlockOwnerDeletion(true)); err != nil {
-		return ctrl.Result{}, fmt.Errorf("failed to set owner reference: %w", err)
-	}
-	if err := cl.Patch(ctx, ent, client.MergeFrom(old)); err != nil {
-		if k8serrors.IsConflict(err) {
-			return ctrl.Result{Requeue: true}, nil
+	if ok, err := controllerutil.HasOwnerReference(consumer.OwnerReferences, ent, cl.Scheme()); err != nil {
+		ctrllog.FromContext(ctx).Info("failed to check if KongConsumer has owner reference", "error", err)
+	} else if ok {
+		old := ent.DeepCopyObject().(TEnt)
```

Comment:
How about leaving some trace of why we want to do it?
```suggestion
		// Make sure there is no ownership between the consumer and the entity by removing
		// a reference if found. Such references may have been added by older versions 
		// of KGO (< 1.5). 
		// TODO: remove this after a couple of minor versions.
		old := ent.DeepCopyObject().(TEnt)
```